### PR TITLE
Allow specifying a dbus socket of tcp:host=localhost

### PR DESCRIPTION
**Pull request by hughsie**
Type of pull request:

- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [ ] Code fix
- [ ] Feature
- [ ] Documentation

### CODE REVIEW

**Review by superm1 — APPROVED** [review]

**Comment by hughsie** [comment]
Screenshot from 2022-05-11 15-35-58

File: libfwupd/fwupd-client.c [hunk 1]
```diff
@@ -818,10 +819,18 @@ fwupd_client_connect_async(FwupdClient *self,
 		return;
 	}
 
-	/* use peer-to-peer only if the env variable is set */
+	/* convert from filename to address, if required */
```
Comment (superm1) [inline libfwupd/fwupd-client.c:822]: I think you want to do something special here for
`#ifdef _WIN32` actually to make fwupd clients work more "automagically".  Maybe look if it's NULL then hardcode to `tcp:...`?

Comment (hughsie) [reply libfwupd/fwupd-client.c:822]: That's a good idea. what port do we use tho? we have to set the same port on the client and the server, and I don't think there's any way to "discover" what port the daemon chose if you leave it to choose something unused.

File: src/fu-main.c [hunk 2]
```diff
@@ -2368,23 +2369,32 @@ main(int argc, char *argv[])
 	}
 #endif
 
-	/* own the object */
+	/* convert from filename to address, if required */
```
Comment (superm1) [inline src/fu-main.c:2372]: similar to my other comment - I think hardcode the daemon to listen on TCP for Windows here.

File: libfwupd/fwupd-client.c (hunk 1 above)
Comment (superm1) [reply libfwupd/fwupd-client.c:822]: How about 1341?  It seems like https://www.iana.org/assignments/service-names-port-numbers/service-names-port-numbers.xhtml?=&skey=2&page=22 it's only used for Qubes today.

**Comment by superm1** [comment]
Can you see if registering a service is basically just this: https://itsiti.com/create-delete-windows-service-using-sc-command/ ?  If so, I think you should register it as part of the MSI.

File: libfwupd/fwupd-client.c (hunk 1 above)
Comment (hughsie) [reply libfwupd/fwupd-client.c:822]: Good idea, fixed.

**Review by superm1 — APPROVED** [review]

**Comment by superm1** [comment]
> @superm1 I think https://wixtoolset.org/documentation/manual/v3/xsd/wix/serviceinstall.html is probably what we want -- no idea if it works. OUTOFTIME for today.

yeah thoughts for the next PR.  This is awesome, great job!

**Pull request merged**